firefox-ios/Client/Frontend/Browser/Tabs/Middleware/TabManagerMiddleware.swift

```diff
@@ -876,11 +876,19 @@ class TabManagerMiddleware: BookmarksRefactorFeatureFlagProvider {
         }
     }
 
-    // MARK: - Homepage Header Actions
-    private func resolveHomepageHeaderActions(with action: HeaderAction) {
+    // MARK: - Homepage Related Actions
+    private func resolveHomepageActions(with action: Action) {
         switch action.actionType {
         case HeaderActionType.toggleHomepageMode:
             tabManager(for: action.windowUUID).switchPrivacyMode()
+        case HomepageActionType.initialize:
+            store.dispatch(
+                TabManagerAction(
+                    recentTabs: tabManager(for: action.windowUUID).recentlyAccessedNormalTabs,
```

Comment:
decided to dispatch an action and have it closely relate to tab manager middleware instead of creating the jumpback in middleware / actions since what powers the jump back in section are the tabs.

Reply:
discussed on slack that this was okay